src/charm.py

```diff
@@ -147,6 +147,20 @@ def _additional_hostnames(self) -> Generator[str, None, None]:
         additional_hostnames = self._get_config_or_relation_data("additional-hostnames", "")
         yield from filter(None, additional_hostnames.split(","))
 
+    @property
+    def _backend_protocol(self) -> str:
+        """Return the backend-protocol to use for k8s ingress."""
+        backend_protocol_default = "HTTP"
+        backend_protocol = self._get_config_or_relation_data(
+            "backend-protocol", backend_protocol_default
+        )
+        # Disabled to reference the documentation
+        # See https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#backend-protocol # noqa: E501 pylint: disable=line-too-long
+        accepted_values = ["HTTP", "HTTPS", "GRPC", "GRPCS", "AJP", "FCGI"]
+        if backend_protocol.upper() not in accepted_values:
+            return backend_protocol_default
+        return f"{backend_protocol}".upper()
```

Comment:
I think this might not need an f-string :D

Reply:
Replaced :)